lib/chilean/rutify.rb

```diff
@@ -9,8 +9,15 @@ module Chilean
   # Chilean rut utils module
   module Rutify
     # checks if the passed value is valid as a rut character
-    def valid_rut_value?(value)
-      ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", "k", "K", ".", "-"].include?(value.to_s)
+    def valid_rut_value?(value, position = 0, length = 1)
+      value_str = value.to_s
+
+      if position == length - 1
+        %w[0 1 2 3 4 5 6 7 8 9 k K].include?(value_str)
+      else
+        # Other positions allow digits, dots, and hyphens
+        ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", ".", "-"].include?(value_str)
+      end
     end
```

Comment:
If we explicitly validate in `valid_rut_values?` it will also make this method a little easier to undestand with less code

```suggestion
   def valid_rut_value?(value, allow_k: false)
      base_values = ["0", "1", "2", "3", "4", "5", "6", "7", "8", "9", ".", "-"]
      base_values.push("K") if allow_k
      base_values.push("k") if allow_k

      base_values.include?(value.to_s)
    end
```